modules/update-upgrading-web.adoc

```diff
@@ -62,6 +62,12 @@ and click *Save*.
 +
 The Input channel
 *Update status* changes to *Update to <product-version> in progress*, and you can review the progress of the cluster update by watching the progress bars for the Operators and nodes.
++
+[NOTE]
+====
+If you are upgrading your cluster to the next minor version, like version 4.y to 4.(y+1), it is recommended to confirm your nodes are upgraded before deploying workloads that rely on a new feature. Any pools with worker nodes that are not yet updated are displayed on the *Cluster Settings* page.
+====
```

Comment:
Seem like this it does not very match with bug reporter's requirement,
"It is an expected behaviour during upgrade process that as soon as MCO verifies that all control planes are upgraded, it marks cluster upgrade complete"

Reply:
A more critical issue is found:
To recreate the same situation, I created a custom mcp, associate one node to this mcp. Start upgrade, the master, worker, and the custom mcp update progress are displayed on the *Cluster Settings* page. The paused mcp was showing 0/1 upgrade progress as expection. After all control planes are upgraded, progress info would disappeared on the page. Even I unpaused the costom mcp, the custom mcp will rolling out upgrade from `oc get mcp` output, but on the *Cluster Settings* page, still no any info is displayed about the left mcp worker upgrade progress. So sound like this change does not help user reported issue.

Reply:
@wking

Reply:
> It is an expected behaviour during upgrade process that as soon as MCO verifies that all control planes are upgraded, it marks cluster upgrade complete

This isn't quite true.  For example, a degraded compute pool will block the MCO from declaring itself updated in its ClusterOperator, which will in turn block the CVO from declaring the update complete in its ClusterVersion object.  I don't think we need to get into those details though.  The wording in this pull request tells folks how to check their compute pools, so they don't need to worry about whether the ClusterVersion object declares the update complete or not.

> And how about moving "Any pools with worker nodes that are not yet updated are displayed on the Cluster Settings page" to a separate item, the sentence is not related to the last sentence, it is easy to be missing for user reading.

Checking the status of compute pools after a minor bump is all that this note is about.  I don't see a benefit into sharding it over multiple paragraphs.

> Even I unpaused the costom mcp, the custom mcp is rolling out upgrade from oc get mcp output, but on the Cluster Settings page, still no any info is displayed about the left mcp worker upgrade progress.

Hmm, that sounds like a separate console bug?  Which should maybe block this bug?  I'd expect that an unpaused pool would lead the MCO to set `Upgradeable=True`, which `oc adm upgrade` will complain about (once the CVO slurps it out of the `machine-config` ClusterOperator and passes it on in ClusterVersion).  Perhaps the console could do something similar?

Reply:
> I'd expect that an unpaused pool would lead the MCO to set Upgradeable=True, which oc adm upgrade will complain about (once the CVO slurps it out of the machine-config ClusterOperator and passes it on in ClusterVersion).

I rerun my testing, after unpause the custom mcp, `oc adm upgrade` report the following message:
```
[root@preserve-jialiu-ansible ~]# oc adm upgrade
Cluster version is 4.9.0-0.nightly-2021-09-08-162532

Upgradeable=False

  Reason: PoolUpdating
  Message: Cluster operator machine-config should not be upgraded between minor versions: One or more machine config pools are updating, please see `oc get mcp` for further details

Upstream is unset, so the cluster will use an appropriate default.
Channel: stable-4.9
warning: Cannot display available updates:
  Reason: VersionNotFound
  Message: Unable to retrieve available updates: currently reconciling cluster version 4.9.0-0.nightly-2021-09-08-162532 not found in the "stable-4.9" channel
```
Meantime, from web console, I can seen the similar warning message on the *Cluster Settings* page, no the left mcp worker upgrade progress is displayed there.

Do you mean that will be enough to notify users about worker upgrade progress, right?

Reply:
> Do you mean that warning message will be enough to notify users about worker upgrade progress, right?

After an hour, [`ClusterNotUpgradeable`][1] will also go off.  But yeah, the purpose of this bug is to make it clear to folks looking at the web console, so bumping the console to be more clear about dangling pools (paused or not), somewhere seems like something that should get addressed, before we point people at the console for that information.

[1]: https://github.com/openshift/cluster-version-operator/blob/c34d95a9e321bed9d10edbf284c452dfc5f6d305/install/0000_90_cluster-version-operator_02_servicemonitor.yaml#L68-L76